node/node_handler.go

```diff
@@ -630,6 +631,10 @@ func (node *Node) PostConsensusProcessing(newBlock *types.Block) {
 			Msg("BINGO !!! Reached Consensus")
 	}
 
+	// mark incomingReceipts as spent and clean up
+	node.Blockchain().MarkCXReceiptsSpent(newBlock)
```

Comment:
Should these be in the insert chain? So it’s more atomic operation?

Reply:
en. good point, let me take a look

Reply:
move this into insertchain